### ISSUE TITLE
Support for Gradient Surgery

### ISSUE THREAD

**Issue by cj-mclaughlin** [open-mmlab/mmsegmentation#1348]
Hello,

I am currently using this library for my experiments, but I would like to have finer control over model training (i.e. scale the magnitude of gradients to specific levels for each layer). 

What is the recommended workflow for something like this? The optimizer code seems to be a part of the base MMCV library. I am hoping to reuse as much as possible here, but please let me know if I am better off re-implementing my own training loop from scratch.

**Comment by MengzhangLI** [comment]
It has to be familiar with MMCV. And use your customized hook/runner rather than default one. 

For example, if you want to make each variable parameter has different learning rate, you could: https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/core/utils/layer_decay_optimizer_constructor.py and set it in config: https://github.com/open-mmlab/mmsegmentation/blob/master/configs/convnext/upernet_convnext_base_fp16_512x512_160k_ade20k.py#L13.

It is inevitable to make your hands dirty to code&debug and read original codes line by line (especially MMCV). No baby sitter instruction till now.

Best,

**Comment by cj-mclaughlin** [comment]
For other users reference, I find a simple approach is to overwrite the forward_train function in the relevant decoder subclasses.
